packages/dendron-next-server/pages/workspace/config.tsx

```diff
@@ -0,0 +1,326 @@
+            disabled={disabled}
+          />
+
+          <FormErrorMessage>{errors.site?.siteRootDir}</FormErrorMessage>
+
+          <FormHelperText>{help}</FormHelperText>
+        </FormControl>
+      )}
+    </Field>
+  );
+}
+
+export default function ConfigSamplePage() {
+  // const { data: configData } = getConfigData();
+
+  const [configData, saveConfigData] = useState(getConfigData().data);
+
+  React.useEffect(() => {
+    console.log("save", JSON.stringify(configData, null, 2));
+  }, [JSON.stringify(configData)]);
+
+  return (
+    <Container paddingBottom={12}>
+      <Stack spacing={4}>
+        <Heading>Config Data</Heading>
+
+        <Formik
+          enableReinitialize
+          initialValues={configData}
+          onSubmit={(formValues) => saveConfigData(formValues)}
+        >
+          {({
+            touched,
+            errors,
+            values,
+            resetForm, // NOTE only works on fields which have configData keys
+          }) => (
+            <Box as={Form} onChange={console.log}>
+              <Stack spacing={8}>
+                <Box as="fieldset">
+                  <Heading size="md" as="legend">
+                    Vaults
+                  </Heading>
+
+                  <Stack spacing={4}>
+                    {configData.vaults.map((vault, index) => (
+                      <Tooltip
+                        label="Vaults are not yet editable via this form."
+                        placement="top"
+                        hasArrow
+                      >
+                        <Stack direction="row" spacing={2}>
+                          <FormLabel
+                            htmlFor={`vaults[${index}].fsPath`}
+                            whiteSpace="nowrap"
+                            paddingTop={2} // align with text field
+                          >
+                            Vault {index}
+                          </FormLabel>
+
+                          <InputControl
+                            name={`vaults[${index}].fsPath`}
+                            placeholder="./filesystemPath"
+                            help="Vault location"
+                            disabled
+                          />
+
+                          <InputControl
+                            name={`vaults[${index}].name`}
+                            placeholder="name"
+                            help="Vault name"
+                            disabled
+                          />
+                        </Stack>
+                      </Tooltip>
+                    ))}
+                  </Stack>
+                </Box>
+
+                <Box as="fieldset">
+                  <Heading size="md" as="legend">
+                    Site
+                  </Heading>
+
+                  <Stack spacing={4}>
+                    <InputControl
+                      label="Site root directory"
+                      name="site.siteRootDir"
+                      placeholder="./docs"
```

Comment:
for placeholder values, can we reference from [DConfig](https://github.com/dendronhq/dendron/blob/master/packages/engine-server/src/config.ts#L14:L14) since these values might change in the future?

Reply:
:heavy_check_mark: @ 0922db6bd7823a646de621435aff2667691a727b -- Stubbed this out, though I think you'll want to augment that getDefaultConfig with more values to avoid blank placeholders when blank equates to the default value:

![Screenshot from 2020-11-22 13-11-23](https://user-images.githubusercontent.com/595711/99917300-3c312d00-2cc4-11eb-930d-e76468d164dd.png)